typescript/opensearch/cwlogs_ingestion/resources/lambda/cw_subscription_filter/handler.py

```diff
@@ -0,0 +1,57 @@
+import base64
+import gzip
+import logging
+import json
+import jmespath
+import requests
+import os
+from datetime import datetime
+from requests_auth_aws_sigv4 import AWSSigV4
+
+def cw_subscription_handler(event, context):
+
+    """Extract the data from the event"""
+    data = jmespath.search("awslogs.data", event)
+
+    """Decompress the logs"""
+    cwLogs = decompress_json_data(data)
+
+    """Construct the payload to send to OpenSearch Ingestion"""
+    payload = prepare_payload(cwLogs)
+
+    """Ingest the set of events to the pipeline"""
+    response = ingestData(payload)
+
+    return {
+        'statusCode': 200,
+        'execute-api': {
+            'status_code': response.status_code,
+            'response': response.text
+        }
+    }
+def decompress_json_data(data):
+    compressed_data = base64.b64decode(data)
+    uncompressed_data = gzip.decompress(compressed_data)
+    return json.loads(uncompressed_data)
+
+def prepare_payload(cwLogs):
+    payload = []
+    logEvents = cwLogs['logEvents']
+    for logEvent in logEvents:
+        request = {}
+        request['@id'] = logEvent['id']
```

Comment:
Why did you include `@` in here? I think this may be confusing and require additional processing to remove. I think we would be fine without the `@`.

Maybe keep it on `@timestamp` if anything.